packages/snaps-simulation/src/methods/hooks/track-error.ts

```diff
@@ -0,0 +1,31 @@
+import { getJsonError } from '@metamask/snaps-sdk';
+import type { SagaIterator } from 'redux-saga';
+import { put } from 'redux-saga/effects';
+
+import type { RunSagaFunction } from '../../store';
+import { trackError } from '../../store';
+
+/**
+ * Track an error.
+ *
+ * @param error - The error to track.
+ * @returns `null`.
+ * @yields Adds the error to the store.
+ */
+function* trackErrorImplementation(error: Error): SagaIterator {
+  const serialisedError = getJsonError(error);
```

Comment:
How come we need this? Shouldn't this mirror the client implementation?

Reply:
In the client we deserialise to an `Error` instance before sending the errors to sentry. I think reserialising makes it easier to test, since you can use a standard object for assertions instead of having to construct an error.